pkg/upgradeconfigmanager/upgradeconfigmanager.go

```diff
@@ -144,9 +144,12 @@ func (s *upgradeConfigManager) StartSync(stopCh context.Context) {
 	}
 
 	duration := durationWithJitter(INITIAL_SYNC_DURATION, JITTER_FACTOR)
+	timeout := time.NewTimer(duration)
+	defer timeout.Stop()
 	for {
+		timeout.Reset(duration)
```

Comment:
why do we need the Reset() here?

Reply:
This is to avoid calling `time.After()` over and over, some of which may not actually trigger and hence not get garbage collected, which causes the leak.
This resets the timer with the updated duration so that it can fire again. That way we are only using one timer object, which is also cleaned up on exit through the earlier `defer`.

Reply:
https://pkg.go.dev/time#Timer.Reset
from the func description here, should we put it to the Reset() after we receive value from channel?

Reply:
That's essentially what happens each time the loop iterates. If it's put at the bottom of the loop we still need to call it before entering the loop in order to set the timer for the first time.
In case I'm not following correctly, can you post a code snippet of how it should work? I have a feeling I'm not getting it.

Reply:
Thanks for the overview here, I've moved the `Reset()` to the bottom of the loop.

Verified operation still working fine after the change, when setting a watch interval of 5 minutes:
```
ts=2021-07-27T04:52:39Z level=info logger=ocm-config-getter msg="Commencing sync with OCM Spec provider"
...
ts=2021-07-27T04:57:50Z level=info logger=ocm-config-getter msg="Commencing sync with OCM Spec provider"
...
ts=2021-07-27T05:02:41Z level=info logger=ocm-config-getter msg="Commencing sync with OCM Spec provider"
...
ts=2021-07-27T05:07:50Z level=info logger=ocm-config-getter msg="Commencing sync with OCM Spec provider"
```